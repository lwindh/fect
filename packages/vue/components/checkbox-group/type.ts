import type { Ref } from 'vue'
import type { NormalSizes } from '../utils'

export const READONLY_CHECKBOX_KEY = Symbol('checkboxKey')

interface CheckboxEeventTarget {
  checked?: boolean
  value?: string[]
}

export interface CheckboxEvent {
  target: CheckboxEeventTarget
  stopPropagation: () => void
  preventDefault: () => void
  nativeEvent: Event
}

export interface CheckboxGroupProvide {
  props: {
    disabled: boolean
    modelValue: string[]
    size: NormalSizes
    useRow: boolean
  }
  parentValue: Ref<string[]>
  updateParentValue: (val: string, checked: boolean) => void
  parentChangeHandler: (e: CheckboxEvent) => void
}
