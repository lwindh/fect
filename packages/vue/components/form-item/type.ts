import type { FormItemProps } from './props'

export const READONLY_FORM_ITEM_KEY = Symbol('FormItemKey')

export interface FormItemProvide {
  formItemProps: FormItemProps
  resetField: () => void
  clearValidate: () => void
}
