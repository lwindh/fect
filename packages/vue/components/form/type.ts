import { tuple } from '../utils'
import type { ComputedRef, Ref } from 'vue'
import type { RuleItem } from 'proy'
import type { FormProps } from './props'

export const labelPosition = tuple('left', 'right', 'top')

export type LabelPosition = typeof labelPosition[number]

export type Trigger = 'change' | 'blur'

export const READONLY_FORM_KEY = Symbol('formKey')

export interface FormProvide {
  formProps: FormProps
  validate: (trigger: Trigger) => void
}

export interface FormRule extends RuleItem {
  message?: string
  trigger?: Trigger
}

export type FormRules = Record<string, FormRule | FormRule[]>
