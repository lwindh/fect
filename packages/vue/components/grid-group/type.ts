import { ComputedRef } from 'vue'
import { tuple } from '../utils'

const justifyTypes = tuple('flex-start', 'center', 'flex-end', 'space-between', 'space-around', 'space-evenly')

const alignTypes = tuple('flex-start', 'center', 'flex-end', 'stretch', 'baseline')

const alignContentTypes = tuple('stretch', 'flex-start', 'center', 'flex-end', 'space-between', 'space-around')

const directionTypes = tuple('row', 'row-reverse', 'column', 'column-reverse')

const wrapTypes = tuple('nowrap', 'wrap', 'wrap-reverse')

export const gridTypes = tuple('xs', 'sm', 'md', 'lg', 'xl')

export type JustifyTypes = typeof justifyTypes[number]

export type AlignTypes = typeof alignTypes[number]

export type AlignContentTypes = typeof alignContentTypes[number]

export type WrapTypes = typeof wrapTypes[number]

export type DirectionTypes = typeof directionTypes[number]

export type GridTypes = typeof gridTypes[number]

export const READONLY_GRID_GROUP_KEY = Symbol('grdiGroup')

export type GridGroupProvide = {
  useGrid: ComputedRef<boolean>
}
