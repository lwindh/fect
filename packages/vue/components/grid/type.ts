import { tuple } from '../utils'

const justifyTypes = tuple(
  'flex-start',
  'center',
  'flex-end',
  'space-between',
  'space-around',
  'space-evenly',
)

const alignTypes = tuple(
  'flex-start',
  'center',
  'flex-end',
  'stretch',
  'baseline',
)

const alignContentTypes = tuple(
  'stretch',
  'flex-start',
  'center',
  'flex-end',
  'space-between',
  'space-around',
)

const directionTypes = tuple('row', 'row-reverse', 'column', 'column-reverse')

const wrapTypes = tuple('nowrap', 'wrap', 'wrap-reverse')

export const sizes = tuple('xs', 'sm', 'md', 'lg', 'xl')

export type JustifyTypes = typeof justifyTypes[number]

export type AlignTypes = typeof alignTypes[number]

export type AlignContentTypes = typeof alignContentTypes[number]

export type WrapTypes = typeof wrapTypes[number]

export type DirectionTypes = typeof directionTypes[number]
