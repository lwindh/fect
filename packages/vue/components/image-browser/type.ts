export type BrowserColors = {
  color: string
  barBgColor: string
  inputBgColor: string
  borderColor: string
  titleColor: string
}
