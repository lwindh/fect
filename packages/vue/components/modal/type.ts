import { Ref } from 'vue'

interface Props {
  visible: boolean
  title: string
  width: string
  cancel: string
  done: string
  teleport: keyof HTMLElementTagNameMap
  overlay: boolean
}

export type ModalProvide = {
  props: Props
  selfVisible: Ref<boolean>
  setSelfVisible: (val: boolean) => void
}

export const READONLY_MODAL_KEY = 'modalKey'
