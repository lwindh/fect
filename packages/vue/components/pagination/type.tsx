import { Ref } from 'vue'
import { NormalSizes } from '../utils'

export interface PaginationSize {
  font: string
  width: string
}

type Props = {
  modelValue: number
  count: string | number
  size: NormalSizes
  prevText: string
  nextText: string
  simple: boolean
  limit: number
}

export type PaginationProvide = {
  setCurrentPage: (val: number) => number
  updateSidePage: (type: string) => void
  head: Ref<boolean>
  end: Ref<boolean>
  props: Props
}

export type SideEvent = 'prev' | 'next'

export const READONLY_PAGINATION_KEY = 'paginationKey'
