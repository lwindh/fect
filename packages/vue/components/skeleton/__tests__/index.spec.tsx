import Skeleton from '../index'
import SkeletonItem from '../../skeleton-item'
import { Variable } from '../../skeleton-item/type'
import { mount } from '@vue/test-utils'

describe('Skeleton', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Skeleton, {
      props: {
        rows: 3,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should render anysome rows and setup width with rowsWidth', async () => {
    const wrapper = mount(Skeleton, {
      props: {
        rows: 3,
        rowsWidth: [10, '80px', '70%'],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setProps({ rowsWidth: '50%' })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should support custom skeleton render', async () => {
    const slots = {
      skeleton: () => <SkeletonItem />,
      default: () => <p>already render</p>,
    }

    const wrapper = mount(Skeleton, {
      slots,
    })
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setProps({
      loading: true,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render different type skeletonItem', () => {
    const types: Variable[] = ['p', 'text', 'h1', 'h3', 'rect', 'circle', 'image', 'button', 'caption']

    const slots = {
      skeleton: () => types.map((type, idx) => <SkeletonItem key={idx} variable={type} />),
    }
    const wrapper = mount(Skeleton, {
      slots,
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('SkeletonItem can not render without Skeleton', () => {
    const wrapper = mount(SkeletonItem)
    expect(() => wrapper.get('div')).toThrowError()
  })
})
