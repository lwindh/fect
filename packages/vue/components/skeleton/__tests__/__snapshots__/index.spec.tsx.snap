// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should be render as a element 1`] = `
<div class="fect-skeleton">
  <div class="skeleton__content">
    <div class="fect-skeleton__item text animated" style="width: 100%;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 100%;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 100%;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Skeleton should render anysome rows and setup width with rowsWidth 1`] = `
<div class="fect-skeleton">
  <div class="skeleton__content">
    <div class="fect-skeleton__item text animated" style="width: 10px;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 80px;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 70%;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Skeleton should render anysome rows and setup width with rowsWidth 2`] = `
<div class="fect-skeleton">
  <div class="skeleton__content">
    <div class="fect-skeleton__item text animated" style="width: 50%;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 50%;">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated" style="width: 50%;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Skeleton should render different type skeletonItem 1`] = `
<div class="fect-skeleton">
  <div class="skeleton__content">
    <div class="fect-skeleton__item p animated">
      <!---->
    </div>
    <div class="fect-skeleton__item text animated">
      <!---->
    </div>
    <div class="fect-skeleton__item h1 animated">
      <!---->
    </div>
    <div class="fect-skeleton__item h3 animated">
      <!---->
    </div>
    <div class="fect-skeleton__item rect animated">
      <!---->
    </div>
    <div class="fect-skeleton__item circle animated">
      <!---->
    </div>
    <div class="fect-skeleton__item image animated"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24" style="color: currentColor;">
        <rect width="18" height="18" x="3" y="3" rx="2" ry="2"></rect>
        <circle cx="8.5" cy="8.5" r="1.5"></circle>
        <path d="M21 15l-5-5L5 21"></path>
      </svg></div>
    <div class="fect-skeleton__item button animated">
      <!---->
    </div>
    <div class="fect-skeleton__item caption animated">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Skeleton should support custom skeleton render 1`] = `
<div class="fect-skeleton">
  <div class="skeleton__content">
    <div class="fect-skeleton__item text animated">
      <!---->
    </div>
  </div>
</div>
`;

exports[`Skeleton should support custom skeleton render 2`] = `<p>already render</p>`;
